client/state/notices/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getSitePost } from 'state/posts/selectors';
+import {
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+export const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: ( action, dispatch, getState ) => {
+		const post = getSitePost( getState(), action.siteId, action.postId );
+
+		let message;
+		if ( post ) {
+			message = translate( 'An error occurred while deleting "%s"', {
+				args: [ truncate( post.title, { length: 24 } ) ]
+			} );
+		} else {
+			message = translate( 'An error occurred while deleting the post' );
+		}
+
+		dispatch( errorNotice( message ) );
+	},
+	[ POST_DELETE_SUCCESS ]: ( action, dispatch ) => {
+		dispatch( successNotice( translate( 'Post successfully deleted' ) ) );
+	},
+	[ POST_SAVE_SUCCESS ]: ( action, dispatch ) => {
+		let text;
+		switch ( action.post.status ) {
+			case 'trash':
+				text = translate( 'Post successfully moved to trash' );
+				break;
+
+			case 'publish':
+				text = translate( 'Post successfully published' );
+				break;
+		}
+
+		if ( text ) {
+			dispatch( successNotice( text ) );
+		}
+	}
+};
+
+export default function noticesMiddleware( { dispatch, getState } ) {
+	return ( next ) => ( action ) => {
```

Comment:
>many of the Redux libraries tend to use fat-arrow functions and without parenthesis for single argument functions

I'm not a huge fan of the single argument exception for just that reason - it's an exception. One which must be learned, and I don't find the benefit to outweigh this cost and the cost of losing a common pattern for argument declarations (i.e. delineated by parentheses). I also don't see it being too much an issue to mix the two styles of functions if it helps bring a good balance of readability and compactness.

I'd agree though that it is a documented style of Redux middleware to take the inlined arrow function approach, and so to meet those expectations it would make more sense to rewrite this similar to how you'd suggested.

I do have strong opinions on the readability of arrow functions generally, but we can save that debate for another day 😄